cloud_utils/cache/stores/redis_store.py

```diff
@@ -0,0 +1,41 @@
+import json
+import logging
+from typing import Callable, Tuple
+
+import redis
+
+
+def _cache_key_name(cache_name: str, key: Tuple) -> str:
+    return f"{cache_name}:{json.dumps(key)}"
+
+
+def make_store(
+    redis_client: redis.Redis,
+    name: str,
+    ttl: int = 0,
+) -> Tuple[Callable, Callable]:
+    def get_item(key: Tuple):
+        result = redis_client.get(_cache_key_name(name, key))
+        if result is None:
+            logging.error(f"{key} is not in {name}")
+            raise KeyError
+        try:
+            return json.loads(result)
+        except ValueError:  # Key contents are malformed (will force key to update).
+            logging.error(f"Malformed key detected: {key} in {name}.")
+            raise KeyError
+
+    def set_item(key: Tuple, value):
+        try:
+            redis_client.setex(
+                _cache_key_name(name, key),
+                ttl,
```

Comment:
i think when ttl=0 redis will expire this. can you check ? if so - we will need an if/else here and use a different command.